sensor/mcp9808/mcp9808.go

```diff
@@ -318,65 +308,77 @@ type Temperature struct {
 	AboveCritical, AboveUpper, BelowLower bool
 }
 
-// AmbientTemp reads the current sensor value along with the flags denoting what boundaries the
-// current temperature exceeds.
-func (d *MCP9808) AmbientTemp() (*Temperature, error) {
-	temp, err := d.Bus.ReadWordFromReg(address, regAmbientTemp)
-	if err != nil {
-		return nil, err
-	}
-
-	tempC := temp &^ 0xF000
-
-	wholeNum := float64(tempC&^0xF) / 16.0
-	fraction := tempC &^ 0xFF0
-
-	tempResult := &Temperature{
-		AboveCritical: !(temp&(1<<15) == 0),
-		AboveUpper:    !(temp&(1<<14) == 0),
-		BelowLower:    !(temp&(1<<13) == 0)}
-	tempResult.CelsiusDeg = wholeNum + (float64(fraction) / 10000.0)
-	if !(temp&(1<<12) == 0) { // read sign bit
-		tempResult.CelsiusDeg *= -1
-	}
-	return tempResult, nil
-}
-
-// readTempC reads from the ambient temperature register and returns the current temperature value in celsius
+// readTempC reads from the reg temperature register and returns the current temperature value in celsius
 func (d *MCP9808) readTempC(reg byte) (float64, error) {
 	temp, err := d.Bus.ReadWordFromReg(address, reg)
 	if err != nil {
 		return 0, err
 	}
 
-	log.Printf("\nTA vs TCrit: %v\nTA vs TUpper: %v\nTA vs TLower: %v\n", !(temp&(1<<15) == 0), !(temp&(1<<14) == 0), !(temp&(1<<13) == 0))
+	return convertWordToTempC(temp), nil
+}
 
-	wholeTempC := temp &^ 0xF000
+func convertWordToTempC(temp uint16) float64 {
```

Comment:
Yes, I realized about the two's complement format as well hence the change for how I handle the fraction `fraction := float64(temp&0xF) * .0625`
I'm not sure how shifting the bit string back and forth like that is simpler or is more accurate than what I have.  Especially since the results I'm getting from the sensor are accurate and after the changes from your first pass I believe are much simpler as well.